src/test/java/de/mediathekview/mserver/crawler/mdr/tasks/MdrFilmTaskTest.java

```diff
@@ -0,0 +1,129 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ForkJoinPool;
+import org.jsoup.Jsoup;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.modules.junit4.PowerMockRunnerDelegate;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Jsoup.class})
+@PowerMockRunnerDelegate(Parameterized.class)
+@PowerMockIgnore("javax.net.ssl.*")
+public class MdrFilmTaskTest extends WireMockTestBase {
+
+  private String requestUrl;
+  private String htmlFile;
+  private String xmlUrl;
+  private String xmlFile;
+  private String expectedTopic;
+  private String expectedTitle;
+  private String expectedDescription;
+  private LocalDateTime expectedTime;
+  private Duration expectedDuration;
+  private String expectedWebsite;
+  private String expectedUrlSmall;
+  private String expectedUrlNormal;
+  private String expectedUrlHd;
+  private String expectedSubtitle;
+  private GeoLocations expectedGeoLocation;
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    return Arrays.asList(
+        new Object[][]{
+            {
+                "https://www.mdr.de/mediathek/fernsehen/sendung803394_date-20180902_inheritancecontext-header_ipgctx-false_numberofelements-1_zc-abd053ed_zs-1638fa4e.html",
+                "/mdr/mdr_film_simple.html",
+                "/mediathek/fernsehen/video-226902-avCustom.xml",
+                "/mdr/mdr_film_info.xml",
+                "Musik & Show",
+                "Ein Abend für Marianne Kiefer",
+                "Als Paula Zipfel und Olga Knopf wurde sie berühmt: Marianne Kiefer. In vielen Lustspielen war die nur 1,54 große Künstlerin die Partnerin von Publikumslieblingen wie Herbert Köfer, Heinz Rennhack oder Ingeborg Krabbe.",
+                LocalDateTime.of(2018, 9, 2, 20, 15, 0),
+                Duration.ofHours(1).plusMinutes(29).plusSeconds(29),
+                "https://www.mdr.de/mediathek/fernsehen/video-226902_zc-7748e51b_zs-1638fa4e.html",
+                "https://odmdr-a.akamaihd.net/mp4dyn2/1/FCMS-1d15c65d-aca1-4c75-88a6-fd6589ad3741-9a4bb04739be_1d.mp4",
+                "https://odmdr-a.akamaihd.net/mp4dyn2/1/FCMS-1d15c65d-aca1-4c75-88a6-fd6589ad3741-730aae549c28_1d.mp4",
+                "https://odmdr-a.akamaihd.net/mp4dyn2/1/FCMS-1d15c65d-aca1-4c75-88a6-fd6589ad3741-be7c2950aac6_1d.mp4",
+                "https://www.mdr.de/mediathek/mdr-videos/b/video-226902-videoSubtitle.xml",
+                GeoLocations.GEO_NONE
+            }
+        });
+  }
+
+  public MdrFilmTaskTest(final String aRequestUrl, final String aHtmlFile, final String aXmlUrl, final String aXmlFile,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 15 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)